src/EnergyPlus/CMakeLists.txt

```diff
@@ -570,7 +570,12 @@ install( TARGETS energyplusapi
   LIBRARY DESTINATION ./ 
 )
 
-if (CMAKE_COMPILER_IS_GNUCXX AND ENABLE_LTO)
+if (MINGW)
```

Comment:
Coincidentally you are very near the trouble I am having getting LTO working again for Linux.  Combination of the link chain and the compiler flags for GNUCXX and LTO.  I'd bet your MinGW build won't LTO build either.